### PR TITLE
layered: split dispatch into no_topo version

### PR DESCRIPTION
**Pull request by JakeHillion**
Refactor layered_dispatch into two functions: layered_dispatch_no_topo and
layered_dispatch. layered_dispatch will delegate to layered_dispatch_no_topo in
the disable_topology case.

Although this code doesn't run when loaded by BPF due to the global constant
bool blocking it, it makes the functions really hard to parse as a human. As
they diverge more and more it makes sense to split them into separate
manageable functions.

This is basically a mechanical change. I duplicated the existing function,
replaced all `disable_topology` with true in `no_topo` and false in the
existing function, then removed all branches which can't be hit.

Test plan:
- Runs on my dev box (6.9.0 fbkernel) with `scx_layered --run-example -n`.
- As above with `-t`.
- CI.

### CODE REVIEW

**Comment by likewhatevs** [comment]
I *think* this is easier to implement if you rebase on #746 btw (also less verifier problems if you're running into those w/ the large boolean expressions in nested loops that pulls out).

**Comment by JakeHillion** [comment]
> I _think_ this is easier to implement if you rebase on #746 btw (also less verifier problems if you're running into those w/ the large boolean expressions in nested loops that pulls out).

Sorry, bit of noise here as this is now a different PR than when it began (out of practice!). All rebased on current `main` and looks fine now.

**Comment by JakeHillion** [comment]
I think this adds some warnings due to unused variables but for ease of verification I'd like to merge it as is, this way it's a very mechanical change without making any adjustments to the code.

File: scheds/rust/scx_layered/src/bpf/main.bpf.c [hunk 1]
```diff
@@ -1214,8 +1214,89 @@ static bool keep_running(struct cpu_ctx *cctx, struct task_struct *p)
...
+		    scx_bpf_consume(layer_idx))
+			return;
+	}
+
+	scx_bpf_consume(LO_FALLBACK_DSQ);
+}
+
 void BPF_STRUCT_OPS(layered_dispatch, s32 cpu, struct task_struct *prev)
 {
+  if (disable_topology)
```
Comment (hodgesds) [inline scheds/rust/scx_layered/src/bpf/main.bpf.c:1297]: just make sure this is formatted properly, otherwise LGTM

**Review by hodgesds — APPROVED** [review]
